challenge_2/go/makernaren/single_number.go

```diff
@@ -0,0 +1,34 @@
+// Take an array of mixed data types and find the element that occurs
+// only once in it.
+package main
+
+import (
+	"fmt"
+)
+
+// Finds the single occurance, takes in interface as input and output params
+//as elements in the given array may be of any data type
+func FindSingle(input []interface{}) (result interface{}) {
+	// Create a hashmap, where element of that array is key and its number of
+	// occurances is the value.
+	occurances := make(map[interface{}]int)
+	for _, key := range input {
+		occurances[key]++
+	}
+	// Return when we find the single occurance.
+	for k, v := range occurances {
+		if v == 1 {
+			result = k
+			break
+		}
+	}
+	return result
```

Comment:
I would recommend adding an additional 'error' return value that can be provided if no result was found.